x-pack/plugins/security_solution/public/actions/add_to_timeline/cell_action/investigate_in_new_timeline.ts

```diff
@@ -38,14 +38,24 @@ export const createInvestigateInNewTimelineCellActionFactory = createCellActionF
       getIconType: () => ADD_TO_TIMELINE_ICON,
       getDisplayName: () => INVESTIGATE_IN_TIMELINE,
       getDisplayNameTooltip: () => INVESTIGATE_IN_TIMELINE,
-      isCompatible: async ({ field }) =>
-        fieldHasCellActions(field.name) && isValidDataProviderField(field.name, field.type),
-      execute: async ({ field, metadata }) => {
+      isCompatible: async ({ data }) => {
+        const field = data[0]?.field;
+
+        return (
+          data.length === 1 && // TODO Add support for multiple values
+          fieldHasCellActions(field.name) &&
+          isValidDataProviderField(field.name, field.type)
+        );
+      },
+      execute: async ({ data, metadata }) => {
+        const field = data[0]?.field;
+        const value = data[0]?.value;
+
```

Comment:
Don't we need  to `if (!field) return;` here?

Reply:
If we trust that `execute` is only called when `isCompatible` returns true. We don't need to check for invalid values.